client/asset/btc/btc.go

```diff
@@ -1687,6 +1687,15 @@ func (btc *ExchangeWallet) AuditContract(coinID, contract, txData dex.Bytes) (*a
 	}, nil
 }
 
+// RefundAddress extracts and returns the refund address from a contract.
+func (btc *ExchangeWallet) RefundAddress(contract dex.Bytes) (string, error) {
```

Comment:
For `*BCHWallet` you'll need to convert with `RecodeCashAddress`. See e.g. `(*BCHWallet).Address()`.